rscripts/run_shiny_server.R

```diff
@@ -0,0 +1,755 @@
+  # read it in
+  return(read_tsv(file_loc, skip = 4))
+}) %>%
+  bind_rows(.id = "samplez")
+
+### Periodicity line plot
+periodicity_df <- lapply(sample_names, function(x) {
+  # path to the read length file
+  file_loc <- paste0(x, "/3nt_periodicity.tsv")
+  # read it in
+  return(read_tsv(file_loc, skip = 4))
+}) %>%
+  bind_rows(.id = "samplez")
+
+### Read counts by frame bar plot
+frame_bar_df <- lapply(sample_names, function(x) {
+  # path to the read length file
+  file_loc <- paste0(x, "/3ntframe_bygene.tsv")
+  # read it in
+  return(read_tsv(file_loc, skip = 4))
+}) %>%
+  bind_rows(.id = "samplez") %>%
+  group_by(samplez) %>%
+  summarise(
+    Frame_0 = sum(Ct_fr0),
+    Frame_1 = sum(Ct_fr1),
+    Frame_2 = sum(Ct_fr2)
+  ) %>%
+  pivot_longer(cols = where(is.numeric),
+               names_to = "Frame",
+               values_to = "Count")
+
+### Read counts by frame boxplot
+frame_box_df <- lapply(sample_names, function(x) {
+  # path to the read length file
+  file_loc <- paste0(x, "/3ntframe_bygene.tsv")
+  # read it in
+  return(read_tsv(file_loc, skip = 4))
+}) %>%
+  bind_rows(.id = "samplez") %>%
+  mutate(
+    Frame_0 = Ct_fr0 / (Ct_fr0 + Ct_fr1 + Ct_fr2),
+    Frame_1 = Ct_fr1 / (Ct_fr0 + Ct_fr1 + Ct_fr2),
+    Frame_2 = Ct_fr2 / (Ct_fr0 + Ct_fr1 + Ct_fr2)
+  ) %>%
+  select(samplez, gene, Frame_0, Frame_1, Frame_2) %>%
+  pivot_longer(names_to = "Frame", cols = 3:5)
+
+### Position specific normalized reads
+pos_sp_df <- lapply(sample_names, function(x) {
+  # path to the read length file
+  file_loc <- paste0(x, "/pos_sp_rpf_norm_reads.tsv")
+  # read it in
+  return(read_tsv(file_loc, skip = 4))
+}) %>%
+  bind_rows(.id = "samplez")
+
+### collated TPMs
+collated_tpms_df <-
+  read_tsv(here::here(yaml$dir_out, "TPMs_collated.tsv"), skip = 4)
+
+### Ribogrid
+ribogrid_df <- lapply(sample_names, function(x) {
+  # path to the read length file
+  file_loc <- paste0(x, "/gene_position_length_counts_5start.tsv")
+  # read it in
+  return(read_tsv(file_loc, skip = 4))
+}) %>%
+  bind_rows(.id = "samplez")
+
+
+### Gene features
+
+# this liekly needs some work, but it depends on the exact format of the feature file which isn't documented yet
+
+if (any(names(yaml) == "features_file")) {
+  if (length(yaml$features_file) > 0) {
+    features_df <- lapply(sample_names, function(x) {
+      file_loc <- paste0(x, "/sequence_features.tsv")
+
+      return(read_tsv(file_loc, skip = 4))
+    }) %>%
+      bind_rows(.id = "samplez")
+
+    possible_features <- unique(features_df$Feature)
+  }
+}
+
+###############################################################################
+# Plotting
+###############################################################################
+
+### define the universal plot theme
+plot_theme <- theme_bw() +
+  theme(
+    panel.background = element_blank(),
+    panel.grid = element_blank(),
+    text = element_text(size = 15)
+  )
+
+### Start the server
+server <- function(input, output, session) {
```

Comment:
Potentially change the name of this function to be more descriptive? I think this function is called on line 755 with line 
`shinyApp(ui, server, options = list(port = port_num))` but I am not entirely sure, so a clearer name would clarify this.

server => create_server

Edit: **I have looked at R documentation for shinyApp and it requires a function called server, so what you have here is fine, I am just unfamiliar with `shiny`